src/glb-director-xdp/bpf/glb_encap.c

```diff
@@ -233,6 +247,11 @@ static __always_inline int xdp_glb_director_process(struct xdp_md *ctx) {
 
 	// cat /sys/kernel/debug/tracing/trace_pipe
 	glb_bpf_printk("Greetings\n");
+
+	uint32_t stat = 0;
+	struct glb_global_stats *g_stats = bpf_map_lookup_elem(&glb_global_packet_counters, &stat);
+	if (g_stats == NULL) return XDP_PASS; /* we must have the stats struct for simplicity */
```

Comment:
Would it be safer to have a dummy `glb_global_packet_counters` to use here or the assumption that this will never actually return `NULL` in practice?

Reply:
> or the assumption that this will never actually return NULL in practice?

Yea, it should never return NULL in this case since it is created automatically at load time. We can't rely on that though, because eBPF needs it checked so there is no possibility of a NULL dereference. I've made the comment clearer.